io-engine/src/bdev/nexus/nexus_child.rs

```diff
@@ -155,14 +152,15 @@ impl Display for Reason {
     }
 }
 
+/// State of a nexus child.
 #[derive(Debug, Clone, Copy, Serialize, Deserialize, Eq, PartialEq)]
 pub enum ChildState {
     /// Child has not been opened, but we are in the process of opening it.
     Init,
     /// Cannot add this block device to the parent as
     /// it iss incompatible property-wise.
     ConfigInvalid,
-    /// The child is open for R/W.
+    /// The child is open for I/O.
```

Comment:
Could this be `Open(SyncState)` ?

Reply:
Faulted can still be out-of-sync. It's independent of being open.

Reply:
As for now, I'd rather think of SyncState as of another dimension of nexus states, independent from Open/Faulted.